src/1_utils.js

```diff
@@ -553,3 +553,74 @@ utils.getBrowserLanguageCode = function() {
 	}
 	return code;
 };
+
+/**
+ * Returns an array which contains the difference in elements between the 'original' and 'toCheck' arrays.
+ * If there is no difference, an empty array will be returned.
+ */
+utils.calculateDiffBetweenArrays = function(original, toCheck) {
+	var diff = [];
+	toCheck.forEach(function(element) {
+		if (original.indexOf(element) === -1) {
+			diff.push(element);
+		}
+	});
+	return diff;
+};
+
+/**
+ * Validates the commerce-data object passed into branch.trackCommerceEvent().
+ * If there are invalid keys present then it will report back what those keys are.
+ * Note: The keys below are optional.
+ */
+utils.validateCommerceDataKeys = function(commerceData) {
+	var allowedInRoot = [ 'common', 'type', 'transaction_id', 'currency', 'revenue', 'revenue_in_usd', 'exchange_rate', 'shipping', 'tax', 'coupon', 'affiliation', 'persona', 'products' ];
+	var allowedInProducts = [ 'sku', 'name', 'price', 'quantity', 'brand', 'category', 'variant' ];
+
+	var invalidKeys = {
+		root: utils.calculateDiffBetweenArrays(allowedInRoot, Object.keys(commerceData)),
+		products: []
+	};
+
+	if (commerceData.hasOwnProperty('products')) {
+		commerceData['products'].forEach(function(product) {
+			invalidKeys['products'] = invalidKeys['products'].concat(utils.calculateDiffBetweenArrays(allowedInProducts, Object.keys(product)));
+		});
+	}
+	return invalidKeys;
+};
+
+/**
+ * Returns an error message if the partner passes in an invalid event or commerce_data to branch.trackCommerceEvent()
+ */
+utils.validateCommerceEventParams = function(event, commerce_data) {
+	if (!event || typeof event !== 'string' || utils.validCommerceEvents.indexOf(event.toLowerCase()) === -1) {
+		return utils.commerceEventMessages.missingPurchaseEvent;
+	}
+
+	if (!commerce_data || typeof commerce_data !== 'object' || Object.keys(commerce_data).length === 0) {
+		return utils.commerceEventMessages.missingCommerceData;
+	}
+
+	var invalidKeys = utils.validateCommerceDataKeys(commerce_data);
```

Comment:
I would rather see this just return the appropriate message string than an object with the the invalid properties. This seems confusing to me